bin/instruments/qmi_bristol_fos

```diff
@@ -0,0 +1,64 @@
+#! /usr/bin/env python3
+
+"""Command line client for the Bristol FOS Fiber-optic switch."""
+
+import argparse
+from contextlib import nullcontext, AbstractContextManager
+import sys
+
+import qmi
+from qmi.instruments.bristol import Bristol_Fos
+from qmi.utils.context_managers import start_stop, open_close
+
+
+def main() -> None:
+
+    parser = argparse.ArgumentParser()
+    parser.description = "Command line client for the Bristol FOS Fiber-optic switch."
+
+    source = parser.add_mutually_exclusive_group(required=True)
+    source.add_argument("--rpc", help="QMI RPC address of device.")
+    source.add_argument("--unique_id", help="Unique ID of the device")
+
+    parser.add_argument("--board_id", type=int, help="Board number for the device. Windows only!")
+    parser.add_argument("--channel", type=int, help="Channel number to set")
+
+    args = parser.parse_args()
+
+    instr: Bristol_Fos
+    with start_stop(qmi, "bristol_fos_client", console_loglevel="WARNING"), parse_source(args) as instr:
```

Comment:
if we don't chain the `with`s, then we can still use the built-in context management protocol here:
```python
with <qmi_context_stuff>:
   if args.unique_id is not None and args.board_id is not None:
       with qmi.make_instrument(
                instrument_name="FOS",
                instrument_class=Bristol_Fos,
                unique_id=args.unique_id,
                board_id=int(args.board_id) as instr:
            instr.select_channel(args.channel)
    elif args.unique_id is not None:
        qmi.make_instrument(
                instrument_name="FOS",
                instrument_class=Bristol_Fos,
                unique_id=args.unique_id
        ) as instr:
            instr.select_channel(args.channel)
... # any other elifs or else
```

Reply:
Yes, that is one, only that it looks a bit more messy than I would like to see with respect to the `parse_args` version. But perhaps we can use the `contextlib` to solve this problem, https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager, by decorating the `parse_args`?

Other option which we have used with the services is the https://docs.python.org/3/library/contextlib.html#contextlib.ExitStack, where we then manually call the `enter_context`.

Reply:
the `contextlib` method looks nice! If you try it out lemme know if it works